src/Symfony/Component/HttpFoundation/AcceptHeader.php

```diff
@@ -147,4 +180,145 @@ private function sort(): void
             $this->sorted = true;
         }
     }
+
+    /**
+     * Sorts candidates by specificity, quality, parameter count, and original order.
+     *
+     * @param AcceptHeaderItem[] $candidates
+     */
+    private function sortCandidates(array &$candidates, AcceptHeaderItem $queryItem): void
+    {
+        usort($candidates, function (AcceptHeaderItem $a, AcceptHeaderItem $b) use ($queryItem) {
+            $specificityA = $this->getSpecificity($a, $queryItem);
+            $specificityB = $this->getSpecificity($b, $queryItem);
+            if ($specificityA !== $specificityB) {
+                return $specificityB <=> $specificityA;
+            }
+            if ($a->getQuality() !== $b->getQuality()) {
+                return $b->getQuality() <=> $a->getQuality();
+            }
+            $parameterCountA = \count($this->getMediaParams($a));
+            $parameterCountB = \count($this->getMediaParams($b));
+            if ($parameterCountA !== $parameterCountB) {
+                return $parameterCountB <=> $parameterCountA;
+            }
+
+            return $a->getIndex() <=> $b->getIndex();
```

Comment:
To all I know, the `<=>` comparison allows to simplify two custom checks into a check on an array of values, with the most important comparison criteria sorted first:
```suggestion
                $comparison = [$this->getSpecificity($b, $queryItem), $b->getQuality(), count($this->getMediaParams($b)] <=> [$this->getSpecificity($a, $queryItem), $a->getQuality(), count($this->getMediaParams($a)];
                if ($comparison !== 0) {
                    return $comparison;
                }
                return $a->getIndex() <=> $b->getIndex();
            }
```